### ISSUE TITLE
Spinner not updating with TypeScript LSP

### ISSUE THREAD

**Issue by icholy** [arkav/lualine-lsp-progress#15]
The spinner doesn't update when using [typescript-language-server](https://github.com/typescript-language-server/typescript-language-server).

![tty](https://user-images.githubusercontent.com/943597/149681533-1b34160f-10d2-48e7-880d-d1bf898114cc.gif)

The LSP doesn't send `$/progress` reports aside from `begin` and `end` (because the underlying tsserver doesn't support them). I proposed sending fake progress reports, but that wasn't accepted. https://github.com/typescript-language-server/typescript-language-server/pull/373

Is there a way to work around this?

**Comment by BobbyGerace** [comment]
I had this issue as well. Here is a pull request with a fix https://github.com/arkav/lualine-lsp-progress/pull/20

**Comment by rockyzhang24** [comment]
The spinner works well in ts file now. Probably some updates on upstream made this issue fixed.

**Comment by icholy** [comment]
@rockyzhang24 the spinner should be updating every 500 milliseconds https://github.com/arkav/lualine-lsp-progress/blob/master/lua/lualine/components/lsp_progress.lua#L25

**Comment by rockyzhang24** [comment]
@icholy Yes it works and updates as expected on my side.

**Comment by icholy** [comment]
@rockyzhang24 which versions of neovim, lualine-lsp-progress, and typescript-langauge-server do you have installed?

**Comment by rockyzhang24** [comment]
@icholy 
neovim version: NVIM v0.8.0-dev+1660-g9e4286294 (the nightly)
lua-lsp-progress: normally installed and the latest commit
typescript-language-server: 0.10.0, the latest

I cloned a large repo `supabase` to make the progress noticeable. 

https://user-images.githubusercontent.com/11582667/168614741-0daa7d3e-d26d-49b9-99f7-2b23b26755c2.mov

**Comment by icholy** [comment]
@rockyzhang24 those are the begin / end progress notifications coming through. If you open a larger project (on a slower computer), you'll notice that it's not updating correctly. Your computer is loading the project so quickly that you're not seeing the issue.

**Comment by rockyzhang24** [comment]
Not really.  They are not just begin and end progress. See the screenshot below.

<img width="1048" alt="1" src="https://user-images.githubusercontent.com/11582667/168627304-4c73ceda-c27b-4c1a-a84f-a21036bd89d8.png">

<img width="1048" alt="2" src="https://user-images.githubusercontent.com/11582667/168627335-47966bb7-9ebc-4790-8c09-3a4b308b640e.png">

<img width="1048" alt="3" src="https://user-images.githubusercontent.com/11582667/168627367-9ab5e3b9-5549-4a05-9ce4-5a44fa034837.png">

Both the first two pics are in the "progress" step. Thanks.

**Comment by icholy** [comment]
I just checked mine again and I am getting 1 update while `(In Progress)`. Again, if you manage to find a project that takes more than a few seconds to load, you will see the issue.

**Comment by rockyzhang24** [comment]
@icholy Let me find a large project and try it again. 

Update:
Yes, you're right. I tested again in a large ts repo and the spinner was stuck after one update while `(In Progress)` until `(Completed)`. 

@arkav Could you please merge the PR to fix this issue? Thanks a lot.

**Comment by icholy** [comment]
@rockyzhang24 out of curiosity, what kind of machine (specs) are you running on?

**Comment by rockyzhang24** [comment]
@icholy 16 inch MacBook Pro with M1 Pro, 32GB Memory + 1TB SSD.
